src/acl/forward.h

```diff
@@ -27,21 +26,31 @@ class AndNode;
 class Answer;
 class ChecklistFiller;
 class InnerNode;
+class NamedRules;
 class NotNode;
 class OrNode;
 class Tree;
 
 /// prepares to parse ACLs configuration
 void Init(void);
 
+// Today, most tree-storing configuration-related classes store a raw pointer to
+// dynamically allocated StoredTree, making explicit aclDestroyAccessList() and
+// aclDestroyAclList() calls for cleanup. TODO: Automate destruction of
+// dynamically-allocated StoredTree objects, eliminating the need for those
+// explicit calls but without exposing storing classes to Acl::Tree details like
+// storing a RefCount<Acl::Tree> pointer would.
+/// reconfiguration-safe storage of ACL rules
+using StoredTree = RefCount<Acl::Tree>;
```

Comment:
> PR description: The outer Config.accessList.foo pointer is still raw because Squid invasive RefCount smart pointers cannot be stored directly in configuration objects without significant out-of-scope refactoring.

We tried really hard to get rid of raw Config.accessList.foo pointers, but several orthogonal attempts had to be abandoned because the changes quickly snowball well beyond this PR scope into noisy changes that polish various configuration-related classes to avoid build errors or undue exposure to ACL internals. The last (and the most promising) unfinished attempt can be seen in branch commit 14621484 that details various problems.

We will come back to these raw pointers. I am sure that the problems we faced are solvable, but we should not try to solve them in one PR. This RP should focus on Acl::Node refcounting changes even if that means keeping Config.accessList.foo and similar pointers raw (but now pointing to smart pointers instead of Acl::Nodes and no longer copied for reuse across reconfiguration).